subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -2276,11 +2276,19 @@ void ticker_job(void *param)
 
 	DEBUG_TICKER_JOB(1);
 
-	/* Defer worker, as job is now running */
+	/* Prevent running job if worker is still running */
 	if (instance->worker_trigger) {
 		DEBUG_TICKER_JOB(0);
 		return;
 	}
+
+	/* Prevent running job if already running (re-entrance) */
```

Comment:
Why is the need for running `ticker_job` from more than one execution/ISR/thread/task context? What is making `ticker_job` to pre-empt?

Reply:
> Why is the need for running `ticker_job` from more than one execution/ISR/thread/task context? What is making `ticker_job` to pre-empt?

We observe it with meta-IRQ threads.

Reply:
Correct my understanding, a meta-IRQ thread should not pre-empt itself, right?

If you agree to the changes here being a workaround for issue with meta-IRQ, then the changes here can be under a workaround Kconfig option.

Reply:
Correct, meta-IRQs run to completion - but pre-empts HCI thread (the debugging is a few months back, so I may need to brush up on the actual flow of things). A workaround config is OK with me, if we can make sure this is not a more general problem.

Reply:
As far as I can tell, the issue occurs when ticker operations are called from thread context and ticker_job is executed inline instead of being chained. An example for this would be a call to ticker_stop from thread context that is preempted by an ISR that then calls, e.g., ticker_update.

Reply:
> ticker operations are called from thread context and ticker_job is executed inline instead of being chained

In this case, TICKER_USER_ID_THREAD is equal to TICKER_USER_ID_ULL_LOW to execute the ticker_job inline.

> ticker_stop from thread context that is preempted by an ISR that then calls, e.g., ticker_update

Here ISR context is not equal to TICKER_USER_ID_ULL_LOW (ISR context has higher priority in comparison to the thread), and hence ticker_job shall not be a inline function when called by ticker_update.

What is important is, there shall only be one thread (or meta-IRQ) calling `mayfly_run(TICKER_USER_ID_ULL_LOW);` in the system.

Reply:
We are investigating whether changing the TICKER_USER_ID_XX "priorities" will solve the problem. This may take a while.

Reply:
@mtpr-ot  I see that I have faced the similar issue to which I propose this: https://github.com/zephyrproject-rtos/zephyr/pull/32752